### PR TITLE
Allow non-ascii characters in requests

### PR DESCRIPTION
**Pull request by schustmi**
## Describe changes
Don't fail if non-ascii characters are in the request body.

## Pre-requisites
Please ensure you have done the following:
- [ ] I have read the **CONTRIBUTING.md** document.
- [ ] If my change requires a change to docs, I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] I have based my new branch on `develop` and the open PR is targeting `develop`. If your branch wasn't based on develop read [Contribution guide on rebasing branch to develop](https://github.com/zenml-io/zenml/blob/main/CONTRIBUTING.md#-pull-requests-rebase-your-branch-on-develop).
- [ ] If my changes require changes to the dashboard, these changes are communicated/requested.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other (add details above)

### CODE REVIEW

File: src/zenml/zen_stores/rest_zen_store.py [hunk 1]
```diff
@@ -275,6 +276,21 @@
...
+    We need to use this when sending data to fastapi, otherwise it fails to
+    parse the request body.
+
+    Args:
+        json_string: A json string
+
+    Returns:
+        The input string with only ascii characters.
+    """
+    return json.dumps(json.loads(json_string), ensure_ascii=True)
```
Comment (strickvl) [inline src/zenml/zen_stores/rest_zen_store.py:291]: CleanShot 2024-08-08 at 13 50 50@2x

Is this what we want? Should we warn people when they're doing this?

Comment (schustmi) [reply src/zenml/zen_stores/rest_zen_store.py:291]: If you load this into a pydantic model, it parses it back into café:

Screenshot 2024-08-08 at 13 54 25

Comment (schustmi) [reply src/zenml/zen_stores/rest_zen_store.py:291]: I also tried with pydantic1, where `model.json()` did the ascii encoding automatically and we never had issues there right?

Comment (strickvl) [reply src/zenml/zen_stores/rest_zen_store.py:291]: Sorry I should have been clearer. I wanted to say that this seems like it'd be unexpected for these values to be transformed this way. Wondering if we want an approach that replaces the letters maybe with some `?` character or something instead?

```python
ascii_text = text.encode('ascii', errors='replace').decode('ascii')
print(ascii_text)  # Output: caf?
```

There's also a bunch of packages that handle this, obv, like `unidecode`. Probably overkill here, but it'd at least turn `café` into `cafe`.

**Review by strickvl — APPROVED** [review]

File: src/zenml/zen_stores/rest_zen_store.py (hunk 1 above)
Comment (schustmi) [reply src/zenml/zen_stores/rest_zen_store.py:291]: This is only for requests we send to our own endpoints, in which the request will get converted to a pydantic model again, which means `café` will stay `café`.

**Review by AlexejPenner — APPROVED** [review]
:seal:

Tested for ö in step comments and in model metadata. Seems to work well.

**Pull request merged**